### PR TITLE
chore: update dbt and dbt_utils version

### PR DESCRIPTION
**Pull request by alfredodimassimo**
[ClickUp Task](https://app.clickup.com/t/85zt9ev77)

This PR updates the dbt_utils version to the most recent 1.1.1 as well as the dbt version for compatibility (per [dbt documentation](https://hub.getdbt.com/dbt-labs/dbt_utils/latest/)). 

### Validation:
Testing the `apply_meta_as_tags` macro
image

### CODE REVIEW

File: dbt_project.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ version: '0.4.0'
 
 config-version: 2
 
-require-dbt-version: ">=1.1.0"
+require-dbt-version: ">=1.3.0"
```
Comment (mbhoopathy) [inline dbt_project.yml:6]: Why 1.3.0?  Is 1.5.0 not compatible?

File: packages.yml [hunk 2]
```diff
@@ -1,3 +1,3 @@
 packages:
   - package: dbt-labs/dbt_utils
-    version: [">=0.7.0", "<1.1.0"]
+    version: 1.1.1
```
Comment (mbhoopathy) [inline packages.yml:3]: Did we find an alternative to `dbt deps`, how did we ensure the package was updated to the specified version?

File: dbt_project.yml (hunk 1 above)
Comment (alfredodimassimo) [reply dbt_project.yml:6]: Yup 1.5.0 would be compatible! 1.3.0 is the minimum required version so I set that as the minimum and any greater version would be acceptable, but I can absolutely change the minimum to 1.5.0.

File: packages.yml (hunk 2 above)
Comment (alfredodimassimo) [reply packages.yml:3]: I did not, I ended up installing dbt and running `dbt deps`

Comment (mbhoopathy) [reply packages.yml:3]: Can you try running the package in a different dbt project, please?
Use can your commit id as the version of snowflake utils package in the other dbt project. 
When it runs successfully, please take a screenshot and add it to the ticket as a means of validation :)

File: dbt_project.yml (hunk 1 above)
Comment (mbhoopathy) [reply dbt_project.yml:6]: ok this means, users need to upgrade to 1.5.0 to use the package. I'll let  @cecile-cecile  weigh on this.

File: packages.yml (hunk 2 above)
Comment (alfredodimassimo) [reply packages.yml:3]: Done! Validation screenshot added into original post :)

File: dbt_project.yml (hunk 1 above)
Comment (mbhoopathy) [reply dbt_project.yml:6]: @alfredodimassimo On retrospect, let's keep 1.3.0 as the minimum version to use the package. We don't want the users to upgrade just to use this package.

File: packages.yml (hunk 2 above)
Comment (mbhoopathy) [reply packages.yml:3]: Great! LGTM :)

Comment (mbhoopathy) [reply packages.yml:3]: one small comment, otherwise LGTM!

**Review by mbhoopathy — APPROVED** [review]

**Pull request merged**